controller/pkg/secrets/cert.go

```diff
@@ -454,3 +436,29 @@ func ensureContainerImageUpdated(container *corev1.Container, imageVersionStr st
 
 	return updated, nil
 }
+
+func parsePrivateKey(pemBlock *pem.Block) (crypto.Signer, x509.SignatureAlgorithm, error) {
+	var (
+		signatureAlgorithm x509.SignatureAlgorithm = x509.UnknownSignatureAlgorithm
+		priv               crypto.Signer
+		err                error
+	)
+	switch pemBlock.Type {
+	case "EC PRIVATE KEY":
+		priv, err = x509.ParseECPrivateKey(pemBlock.Bytes)
+		if err != nil {
+			return nil, signatureAlgorithm, err
+		}
+		signatureAlgorithm = x509.ECDSAWithSHA256
```

Comment:
Nit: initializing the variables above could be avoided if we had a direct return in every case.
```suggestion
		return priv, x509.ECDSAWithSHA256, nil
```